### PR TITLE
Low thrust costates

### PR DESCRIPTION
**Pull request by DominicDirkx**
Code to determine direction of a thrust force, based on Modified Equinoctial Element costates. Code strongly based on work by @FrankHogervorst.

### CODE REVIEW

**Comment by DominicDirkx** [comment]
Hi Frank, 

Would you have the time to have a look through this code? Most of it is largely unchanged from your implementation,

Cheers,

Dominic

**Comment by FrankHogervorst** [comment]
Yes, I will try to do it before the weekend.

**Comment by DominicDirkx** [comment]
Great, no rush, I want to be able to provide it to the P&O students in about 2 weeks. We can also quickly discuss it tomorrow, depending on how many cool figures you have to show :)

File: Tudat/Astrodynamics/Propulsion/costateBasedThrustGuidance.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      costateFunction_( costateFunction ){ }
+
+//! Function to update the force direction to the current time.
+void MeeCostateBasedThrustGuidance::updateForceDirection( const double time )
+{
+    if( !( time == currentTime_ ) )
+    {
+        Eigen::VectorXd costates_ = costateFunction_( time );
+
+        // Get the current state in cartesian coordinates and keplerian elements, and some convenient parameters
```
Comment (FrankHogervorst) [inline Tudat/Astrodynamics/Propulsion/costateBasedThrustGuidance.cpp]: No keplerian elements are obtained

File: Tudat/Astrodynamics/Propulsion/costateBasedThrustGuidance.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      centralBodyStateFunction_( centralBodyStateFunction ),
+      centralBodyGravitationalParameterFunction_( centralBodyGravitationalParameterFunction ),
+      costateFunction_( costateFunction ){ }
+
+//! Function to update the force direction to the current time.
+void MeeCostateBasedThrustGuidance::updateForceDirection( const double time )
+{
+    if( !( time == currentTime_ ) )
+    {
+        Eigen::VectorXd costates_ = costateFunction_( time );
```
Comment (FrankHogervorst) [inline Tudat/Astrodynamics/Propulsion/costateBasedThrustGuidance.cpp]: I would add a check for the length of the costates vector, as it is assumed that it has length 5.

**Pull request merged**